### PR TITLE
Add support for (un)flagging posts + listing flagged posts

### LINKED ISSUE #50: Add Flag functionality for Newsfeed
In order to make the Newsfeed truly usable attendees need to be able to flag offensive posts and allow moderators (through the Organizer app) to delete them.

This would most likely involve the following:
- Adding a current_user_flagged (to work much like current_user_likes so they can't flag it multiple times
- Adding a num_flagged to tell how many users have flagged the post (Organizers could sort by most flagged)
- Adding API calls to flag it of unflag it (from attendee perspective (much like likes)
- Adding API request to fetch all that have been flagged (in organizer app)
- Adding API request to delete post if moderator deems it inappropriate (in organizer app)

### PR DESCRIPTION
**Pull request by kyl191**
Resolves #50

### CODE REVIEW

File: api/models/social.py [hunk 1]
```diff
@@ -28,6 +28,10 @@
 EXT_POSTS_TOPIC = 'arn:aws:sns:us-west-2:037954390517:calligre-external-posts'
 DEFAULT_PROFILE_PIC = 'https://s3-us-west-2.amazonaws.com/'
 'calligre-profilepics/default.png'
+POSTS_TABLE_NAME = os.environ.get('POSTS_TABLE', 'calligre-posts')
+
+posts_table = dynamo.DynamoWrapper(table_name=POSTS_TABLE_NAME)
+flag_table = dynamo.DynamoWrapper(table_name='flagged')
```
Comment (TheKevJames) [inline api/models/social.py]: May as well make this an env var as well

File: api/models/social.py [hunk 2]
```diff
@@ -73,8 +77,11 @@ def format_post_response(posts, req_userid):
         item['poster_name'] = res.get(item['poster_id'], {}).get('name')
         item['poster_icon'] = res.get(item['poster_id'], {}).get('poster_icon')
         item['current_user_likes'] = req_userid in item.get('likes', [])
-        item['like_count'] = str(item.get('like_count'))
+        item['like_count'] = str(item.get('like_count'), 0)
```
Comment (TheKevJames) [inline api/models/social.py]: Pretty sure your brackets are wrong here. Don't you want: `str(item.get('like_count', 0))`?

File: api/models/social.py [hunk 3]
```diff
@@ -73,8 +77,11 @@ def format_post_response(posts, req_userid):
...
+        item['current_user_flagged'] = req_userid in item.get('flags', [])
+        item['flag_count'] = str(item.get('flag_count'), 0)
```
Comment (TheKevJames) [inline api/models/social.py]: Pretty sure your brackets are wrong here. Don't you want: `str(item.get('flag_count', 0))`?

File: api/models/social.py [hunk 4]
```diff
@@ -412,4 +433,140 @@ def post(self, postid):
...
+            return r, status
+
+        # unconditionally create the flag record because we're only
+        # overwritting the parition key attr, which will always be the same
+        params = {
+            'Item': {
+                'timestamp': timestamp,
+            },
+        }
+        flag_table.put(params)
```
Comment (TheKevJames) [inline api/models/social.py]: Do we need error/response handling here?

**Review by TheKevJames — CHANGES_REQUESTED** [review]

File: api/models/social.py (hunk 4 above)
Comment (kyl191) [reply api/models/social.py]: Both `patch` and `put` return a 204 on success. 

I've simplified it to just using the return value from the put.

**Review by TheKevJames — APPROVED** [review]

**Pull request merged**